File: app/src/main/java/com/github/pwittchen/gesture/app/GestureActivity.java
package com.github.pwittchen.gesture.app;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.Menu;
import android.view.MenuItem;
import android.view.MotionEvent;
import android.widget.TextView;
import com.github.pwittchen.gesture.library.Gesture;
import com.github.pwittchen.gesture.library.GestureListener;

public class GestureActivity extends AppCompatActivity {
  private Gesture gesture;
  private TextView textView;

  @Override public void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.main);
    textView = (TextView) findViewById(R.id.textView);
    gesture = new Gesture();
    gesture.addListener(new GestureListener() {
      @Override public void onPress(MotionEvent motionEvent) {
        textView.setText("press");
      }

      @Override public void onTap(MotionEvent motionEvent) {
        textView.setText("tap");
      }

      @Override public void onDrag(MotionEvent motionEvent) {
        textView.setText("drag");
      }

      @Override public void onMove(MotionEvent motionEvent) {
        textView.setText("move");
      }

      @Override public void onRelease(MotionEvent motionEvent) {
        textView.setText("release");
      }

      @Override public void onLongPress(MotionEvent motionEvent) {
        textView.setText("longpress");
      }

      @Override public void onMultiTap(MotionEvent motionEvent, int clicks) {
        textView.setText("multitap [" + clicks + "]");
      }
    });
  }

  @Override public boolean dispatchTouchEvent(MotionEvent event) {
    gesture.dispatchTouchEvent(event);
    return super.dispatchTouchEvent(event);
  }

  @Override public boolean onCreateOptionsMenu(Menu menu) {
    getMenuInflater().inflate(R.menu.main_menu, menu);
    return true;
  }

  @Override public boolean onOptionsItemSelected(MenuItem item) {
    switch (item.getItemId()) {
      case R.id.listener:
        break;
      case R.id.rx:
        final Intent intent = new Intent(this, GestureRxActivity.class);
        startActivity(intent);
        break;
    }
    return true;
  }
}

File: app-kotlin/src/main/kotlin/com/github/pwittchen/gesture/kotlinapp/GestureActivity.kt
package com.github.pwittchen.gesture.kotlinapp

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.Menu
import android.view.MenuItem
import android.view.MotionEvent
import com.github.pwittchen.gesture.library.Gesture
import com.github.pwittchen.gesture.library.GestureListener
import kotlinx.android.synthetic.main.main.textView

class GestureActivity : AppCompatActivity() {
  private var gesture: Gesture? = null
  public override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.main)

    gesture = Gesture()

    (gesture as Gesture).addListener(object : GestureListener {
      override fun onPress(motionEvent: MotionEvent) {
        textView.text = "press"
      }

      override fun onTap(motionEvent: MotionEvent) {
        textView.text = "tap"
      }

      override fun onDrag(motionEvent: MotionEvent) {
        textView.text = "drag"
      }

      override fun onMove(motionEvent: MotionEvent) {
        textView.text = "move"
      }

      override fun onRelease(motionEvent: MotionEvent) {
        textView.text = "release"
      }

      override fun onLongPress(motionEvent: MotionEvent) {
        textView.text = "longpress"
      }

      override fun onMultiTap(motionEvent: MotionEvent, clicks: Int) {
        textView.text = "multitap [$clicks]"
      }
    })
  }

  override fun dispatchTouchEvent(event: MotionEvent): Boolean {
    gesture!!.dispatchTouchEvent(event)
    return super.dispatchTouchEvent(event)
  }

  override fun onCreateOptionsMenu(menu: Menu): Boolean {
    menuInflater.inflate(R.menu.main_menu, menu)
    return true
  }

  override fun onOptionsItemSelected(item: MenuItem): Boolean {
    when (item.itemId) {
      R.id.listener -> {
      }
      R.id.rx -> {
        val intent = Intent(this, GestureRxActivity::class.java)
        startActivity(intent)
      }
    }
    return true
  }
}